inbm-lib/inbm_common_lib/utility.py

```diff
@@ -124,7 +124,7 @@ def remove_file(path: Union[str, Path]) -> bool:
         return False
 
 
-def create_file_with_contents(path: Union[str, Path], contents: List[str]) -> None:
+def create_file_with_contents(path: Union[str, Path], contents: list[str]) -> None:
     """ Create a file and add the contents by line
```

Comment:
**<h2>$\color{#15F4EE}Miscellaneous$</h2>**
**Suggestion:** Replace the direct string concatenation with os.path.join to construct file paths. This ensures that the file paths are constructed correctly across different operating systems.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
canonical_path = get_canonical_representation_of_path(str(path))
with open(canonical_path, 'w') as f:
    f.writelines([string + os.linesep for string in contents])
```